frontend/src/pages/modelCatalog/routeUtils.ts

```diff
@@ -0,0 +1,11 @@
+import { encodeParams } from './utils';
+
+export const modelCatalogUrl = (): string => `/modelCatalog`;
+
+export const modelDetailsUrl = (params: Record<string, string>): string => {
```

Comment:
instead of `Record<string, string>`, can you use the types you'd expect to use here for this object (include sourceName, repositoryName, etc...) and lift that into an exported type `ModelDetailsRouteParams`? You'll need to either have all of the properties of that type optional, or use `Partial<ModelDetailsRouteParams>` in your function. See my next comment...